### PR TITLE
Only start session if the session has not been started before

### PR DESCRIPTION
**Pull request by peterrehm**
As discussed in #731.

### CODE REVIEW

File: tests/SessionTest.php [hunk 1]
```diff
@@ -83,34 +101,10 @@ public function testRestart()
         $this->session->restart();
     }
 
-    public function testVisitWithoutRunningSession()
```
Comment (stof) [inline tests/SessionTest.php]: this test must not be removed. We still need to cover the case of calling ``visit`` on a started and on a non-started session IMO (thus, your mocks don't describe the actual interaction happening, as you removed the mocking of ``isStarted``

File: src/Session.php [hunk 2]
```diff
@@ -67,7 +67,10 @@ public function isStarted()
      */
     public function start()
     {
-        $this->driver->start();
+        // start session if needed
```
Comment (stof) [inline src/Session.php]: useless comment. The phpdoc already says it

File: src/Session.php [hunk 3]
```diff
@@ -55,10 +55,12 @@ public function isStarted()
     }
 
     /**
-     * Starts session driver.
+     * Starts session driver if session is not started.
      *
      * Calling any action before visiting a page is an undefined behavior.
      * The only supported method calls on a fresh driver are
+     * - resizeWindow()
+     * - maximizeWindow()
```
Comment (stof) [inline src/Session.php]: if we want to allow this, we must add new tests in the driver testsuite calling these methods before visiting a page, to ensure it actually works in all drivers

Comment (stof) [inline src/Session.php]: btw, this does not seem related to this PR, but to #733

Comment (peterrehm) [inline src/Session.php]: Correct, that is my bad. Where would you add the tests? Looking at the CoreDriverTest and the Tests in the Selenium driver I could not see any tests on e.g. the visit method

Comment (peterrehm) [inline src/Session.php]: I have moved that to #733.

Comment (stof) [inline src/Session.php]: the driver testsuite lives in https://github.com/minkphp/driver-testsuite and is a dev dependency in all drivers repositories, to share the same testsuite in all of them.

Comment (peterrehm) [inline src/Session.php]: That makes sense. I was scanning the testsuite, I could not find a test where the autostart is tested e.g. for the visit method by grepping te repository. Do you know if there is anything in this regard?

Comment (stof) [inline src/Session.php]: Allowing to call these methods without visiting a page is **not** related to auto-starting. It is a different feature (and this is what you need to test)

Comment (peterrehm) [inline src/Session.php]: Oh okay. So what do you think about: https://github.com/minkphp/driver-testsuite/pull/10

File: tests/SessionTest.php (hunk 1 above)
Comment (aik099) [inline tests/SessionTest.php]: Actually the discussion from https://github.com/minkphp/Mink/pull/733/files#r99773293 is meant to happen on this PR instead.
